src/main/java/com/conveyal/osmlib/OSM.java

```diff
@@ -181,6 +181,7 @@ public void readFromFile(String filePath) {
                 // FIXME this takes two minutes on NL OSM. We should probably save the intersections in a MapDB table.
                 LOG.info("Detecting intersections...");
                 for (Way way : ways.values()) {
+                    if (way.hasTag("building")) continue;
```

Comment:
Oh, so you're only skipping the buildings on intersection detection. I was thinking we should skip them entirely when loading the PBF, that way we also save time and space saving building ways that we'll never use.